godot-macros/src/class/derive_godot_class.rs

```diff
@@ -253,6 +253,73 @@ fn make_godot_init_impl(class_name: &Ident, fields: &Fields) -> TokenStream {
     }
 }
 
+fn make_onready_init(all_fields: &[Field]) -> TokenStream {
+    let mut onready_fields = all_fields
+        .iter()
+        .filter(|&field| field.is_onready)
+        .map(|field| {
+            let field = &field.name;
+            quote! {
+                ::godot::private::auto_init(&mut self.#field, &base);
+            }
+        });
+
+    if let Some(first) = onready_fields.next() {
+        quote! {
+            {
+                let base = <Self as godot::obj::WithBaseField>::to_gd(self).upcast();
+                #first
+                #( #onready_fields )*
+            }
+        }
+    } else {
+        TokenStream::new()
+    }
+}
+
+fn make_oneditor_panic_inits(class_name: &Ident, all_fields: &[Field]) -> TokenStream {
+    // Despite its name OnEditor shouldn't panic in the editor for tool classes.
+    let editor_check = quote! { ::godot::classes::Engine::singleton().is_editor_hint() };
+
+    // Inform the user which fields haven't been set, instead of panicking on the very first one. Useful for debugging.
+    let on_editor_fields_checks = all_fields
+        .iter()
+        .filter(|&field| field.is_oneditor)
+        .map(|field| {
+            let field = &field.name;
+            let warning_message =
+                format! { "godot-rust: OnEditor field {field} hasn't been initialized."};
+            quote! {
+            if this.#field.is_invalid() {
+                ::godot::global::godot_warn!(#warning_message);
+                is_oneditor_properly_initialized = false;
+            }
+            }
+        })
+        .collect::<Vec<_>>();
+
+    if !on_editor_fields_checks.is_empty() {
+        quote! {
+            fn __are_oneditor_fields_initalized(this: &#class_name) -> bool {
+                if #editor_check {
+                    return true;
+                }
+
+                let mut is_oneditor_properly_initialized: bool = true;
+                #( #on_editor_fields_checks )*
+
+                is_oneditor_properly_initialized
```

Comment:
Also here -- "properly" sounds vague and sloppy; if there isn't a more precise name, at least a comment should be used to clarify.